core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/Experiment.java

```diff
@@ -12,234 +12,369 @@
-   *
-   * @return String accession.
-   */
-  public String getAccession();
+  private String description;
 
-  /**
-   * Sets the accession of this Experiment object.
-   *
-   * @param accession accession.
-   *
-   */
-  public void setAccession(String accession);
+  @Id
+  @GeneratedValue(strategy = GenerationType.AUTO)
+  private long experimentId = UNSAVED_ID;
 
-  /**
-   * Returns the alias of this Experiment object.
-   *
-   * @return String alias.
-   */
-  public String getAlias();
+  @ManyToMany(targetEntity = KitImpl.class)
+  @JoinTable(name = "Experiment_Kit", inverseJoinColumns = { @JoinColumn(name = "kits_kitId") }, joinColumns = {
+      @JoinColumn(name = "experiments_experimentId") })
+  private Collection<Kit> kits = new HashSet<>();
 
-  /**
-   * Sets the alias of this Experiment object.
-   *
-   * @param alias alias.
-   *
-   */
-  public void setAlias(String alias);
+  @ManyToOne(targetEntity = UserImpl.class)
+  @JoinColumn(name = "lastModifier")
+  private User lastModifier;
 
-  /**
-   * Returns the title of this Experiment object.
-   *
-   * @return String title.
-   */
-  public String getTitle();
+  // defines a library on which this experiment will operate.
+  @ManyToOne(targetEntity = LibraryImpl.class)
+  @JoinColumn(name = "library_libraryId")
+  private Library library;
 
-  /**
-   * Sets the title of this Experiment object.
-   *
-   * @param title title.
-   *
-   */
-  public void setTitle(String title);
+  @Column(nullable = false)
+  private String name;
 
-  /**
-   * Returns the name of this Experiment object.
-   *
-   * @return String name.
-   */
-  public String getName();
+  @ManyToOne(targetEntity = Platform.class)
+  @JoinColumn(name = "platform_platformId")
+  private Platform platform;
 
-  /**
-   * Sets the name of this Experiment object.
-   *
-   * @param name name.
-   *
-   */
-  public void setName(String name);
+  // defines the parent run which processes this experiment
+  @OneToMany(mappedBy = "experiment", cascade=CascadeType.ALL)
+  private List<RunPartition> runPartitions;
 
-  /**
-   * Returns the description of this Experiment object.
-   *
-   * @return String description.
-   */
-  public String getDescription();
+  @ManyToOne(cascade = CascadeType.PERSIST)
+  @JoinColumn(name = "securityProfile_profileId")
+  private SecurityProfile securityProfile = new SecurityProfile();
 
-  /**
-   * Sets the description of this Experiment object.
-   *
-   * @param description description.
-   *
-   */
-  public void setDescription(String description);
+  @ManyToOne(targetEntity = StudyImpl.class)
+  @JoinColumn(name = "study_studyId")
+  private Study study;
+  @Column(nullable = false)
+  private String title;
 
   /**
-   * Returns the platform of this Experiment object.
-   *
-   * @return Platform platform.
+   * Construct a new Experiment with a default empty SecurityProfile
    */
-  public Platform getPlatform();
+  public Experiment() {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)